compiler/cpp/CMakeLists.txt

```diff
@@ -113,7 +113,7 @@ THRIFT_ADD_COMPILER(xml     "Enable compiler for XML" ON)
 # we also add the current binary directory for generated files
 include_directories(${CMAKE_CURRENT_BINARY_DIR} src)
 
-if(NOT ${WITH_PLUGIN})
+if(NOT DEFINED WITH_PLUGIN OR NOT ${WITH_PLUGIN})
```

Comment:
This should not be necessary, as https://github.com/apache/thrift/blob/master/build/cmake/DefineOptions.cmake#L97 ensures it is always defined.  Perhaps you just need to specify `-DWITH_PLUGIN=OFF` ?

Reply:
Oh, I know, because I build the compiler under `compiler/cpp/` and  `DefineOptions.cmake` is at `build/cmake`, so `WITH_PLUGIN` is not defined?

Reply:
I see, I always generate the make environment from the top level, then if I only need the compiler I will cd into the build directory's compiler/cpp and make from there.